src/fu-util.c

```diff
@@ -573,6 +574,28 @@ fu_util_print_device_tree (GNode *n, gpointer data)
 	return FALSE;
 }
 
+static gboolean
+fu_util_remove_extra_devices (FuUtilPrivate *priv, GPtrArray *devs, GError **error)
+{
+	if (priv->show_all_devices)
+		return TRUE;
+
+	for (guint i = 0; i < devs->len; i++) {
+		FwupdDevice *dev_tmp = g_ptr_array_index (devs, i);
+		if (fwupd_device_has_flag (dev_tmp, FWUPD_DEVICE_FLAG_UPDATABLE))
+			continue;
+		if (!g_ptr_array_remove (devs, dev_tmp)) {
```

Comment:
I don't think it's safe to carry on iterating the list when you've removed an element. Normally you just return a new list which has only the elements you require added to it.

Reply:
Ah, I thought that glib was abstracting this risk, but it's not.  I've moved it to just do the check in place for get-devices and get-topology.  get-updates implicitly doesn't benefit from it.